src/vignette/util/thumb_verifier.clj

```diff
@@ -0,0 +1,169 @@
+(ns vignette.util.thumb-verifier
+  (:require [wikia.common.logger :as log]
+            [clojure.java.shell :refer [sh]])
+  (:use [environ.core]))
+
+(declare
+  thumb-size-estimators
+  estimate-thumb-size
+  size-of)
+
+(def thumb-size-error "thumbnail-size")
+
+(def thumb-verification-error "thumbnail-verification")
+
+(defn check-thumb-size
+  "Checks if the generated thumbnail has the correct size.
+  If not, an error is logged."
+  [thumb-map thumb original]
+  (try
+    (if-let [estimator
+              (thumb-size-estimators (keyword (:thumbnail-mode thumb-map)))]
+      (let [thumb-size (size-of thumb)
+            estimated-size (estimate-thumb-size estimator thumb-map original)]
+              (if (not= estimated-size thumb-size)
+                (log/error "Thumbnail size is incorrect!" {
+                  :type thumb-size-error
+                  :thumb-map thumb-map
+                  :estimated estimated-size
+                  :actual thumb-size
+                  })))
+      (log/error "Couldn't verify thumbnail size. No estimator found." {
+        :type thumb-verification-error
+        :thumb-map thumb-map
+        }))
+    (catch Exception e (log/error "Thumbnail verification failed" {
+      :type thumb-verification-error
+      :thumb-map thumb-map
+      }))))
+
+(def identify-bin (str (env :imagemagick-base "/usr/local/bin") "/identify"))
+
+(defn identify
+  [file]
+  (:out (sh identify-bin file)))
+
+(defn size-of
```

Comment:
I think you can try using Apache Tika instead of shelling out to imagick-identify